### PR TITLE
Added special option 'encode' to `$htmlOptions` argument in `CHtml::errorSummary` and `CHtml::error`

### PR DESCRIPTION
**Pull request by shidenko97**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Tests pass?   | ✔️
| Fixed issues  | #4317

### CODE REVIEW

File: framework/web/helpers/CHtml.php [hunk 1]
```diff
@@ -2220,7 +2224,11 @@ public static function errorSummary($model,$header=null,$footer=null,$htmlOption
 				foreach($errors as $error)
 				{
 					if($error!='')
-						$content.= '<li>'.$error."</li>\n";
+					{
+                        if (!isset($htmlOptions['ignoreHtmlEncode']) || !$htmlOptions['ignoreHtmlEncode'])
```
Comment (samdark) [inline framework/web/helpers/CHtml.php:2228]: Let's be consistent in naming: https://github.com/yiisoft/yii/blob/master/framework/web/helpers/CHtml.php#L153

Comment (shidenko97) [reply framework/web/helpers/CHtml.php:2228]: Sorry, have not seen it. Few moments and I will fix it with tabs.

File: framework/web/helpers/CHtml.php [hunk 2]
```diff
@@ -2197,6 +2197,10 @@ public static function activeRadioButtonList($model,$attribute,$data,$htmlOption
 	 * make the error summary to show only the first error message of each attribute.
 	 * If this is not set or is false, all error messages will be displayed.
 	 * This option has been available since version 1.1.3.
+	 * Another special option named 'encode' is recognized, which when set true, will
```
Comment (samdark) [inline framework/web/helpers/CHtml.php:2200]: It should be `false`:

```suggestion
	 * Another special option named 'encode' is recognized, which when set false, will
```

File: framework/web/helpers/CHtml.php [hunk 3]
```diff
@@ -2197,6 +2197,10 @@ public static function activeRadioButtonList($model,$attribute,$data,$htmlOption
...
+	 * disable the CHtml::encode encoding of all error messages.
+	 * If this is not set or is false, all error messages will be encoded by CHtml::encode.
```
Comment (samdark) [inline framework/web/helpers/CHtml.php:2202]:
```suggestion
	 * If this is not set or is true, all error messages will be encoded by CHtml::encode.
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/web/helpers/CHtml.php (hunk 2 above)
Comment (shidenko97) [reply framework/web/helpers/CHtml.php]: Thanks, I've changed it and condition to correct

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**